### ISSUE TITLE
BigInt literal incorrectly created from int32

### ISSUE THREAD

**Issue by ncave** [fable-compiler/Fable#1623]
BigInt is implemented, but the literal doesn't seem to work:

- bigint.Parse "12345678912345" works and gives 12345678912345
- but 12345678912345I gives 1942903641

**Comment by alfonsogarciacaro** [comment]
This seems to happen only in the REPL, it works with dotnet-fable CLI. I think we had a similar issue recently with number parsing in the REPL #1539

**Comment by alfonsogarciacaro** [comment]
The AST provided by the F# compiler [uses `FromInt64` for `12345678912345I`](http://ratatosk.dynu.net/ast/#?code=DYUwLgBAlhC8EEYBMBmALAVgGwHYAcAnMuhgJJA) while Fable REPL uses `fromInt32` which suggests there's an integer truncation happening somewhere (it works for numbers within the int32 range).

Another problem is that for large literals (e.g. `999999999999999999999999999999999I`) the F# compiler calls `BigInteger.FromString` while the Fable REPL calls `Bigint/fromZero()`.

Is this related with [big number literal parsing](https://github.com/ncave/FSharp.Compiler.Service/blob/cbd257823bcdda8289d9c03ba34affca598e3b41/src/fsharp/lex.fsl#L394-L396)? Do we need to create a constant in Fable REPL as we do for longs and decimals?

**Comment by ncave** [comment]
@alfonsogarciacaro No, it's related to int32 and int64 parsing which doesn't properly check the range.
I know what to fix but was waiting for your decimal change first, as you have quite a few changes in there.
